### PR TITLE
boards: esp32c3_devkitc: Added board

### PR DESCRIPTION
**Pull request by raffarost**
Support for board esp32c3_devkitc

### CODE REVIEW

**Comment by kartben** [comment]
it looks to me as the two commits should probably be two different PRs?

**Review by uLipe — CHANGES_REQUESTED** [review]
As @kartben mentioned, there are two commits and one of them is unrelated to the PR.

Please split this PR into two separated one. Optionally you can open and merge a PR for the first patch, and make this one blocked by it.

**Comment by raffarost** [comment]
Sorry I still didn't grasp the best practices. It was not thoughtful to include other boards in the PR.
However to me it makes sense to keep the second commit limited to esp32c3, as the idea is to render all samples moved from `boards` to `socs` available to the new board supported (esp32c3_devkitc). Is that okay?

**Review by LucasTambor — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by uLipe — APPROVED** [review]

File: boards/espressif/esp32c3_devkitc/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# ESP32C3 devkitc board configuration
+
+# Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 65535 if WIFI && BT
```
Comment (kartben) [inline boards/espressif/esp32c3_devkitc/Kconfig.defconfig:8]: fwiw other esp32 boards have recently been updated to:

```suggestion
	default $(UINT16_MAX) if WIFI && BT
```

not sure this was fully intended though, as it looks to me as if this is a reasonably arbitrary figure that just happens to be equal to UINT16_MAX, not really something that's semantically *meant* to be UINT16_MAX

either way, it should probably be consistent across esp32 boards

Comment (marekmatej) [reply boards/espressif/esp32c3_devkitc/Kconfig.defconfig:8]: A more suitable macro here would be `KB(64)`   :)

Comment (sylvioalves) [reply boards/espressif/esp32c3_devkitc/Kconfig.defconfig:8]: @kartben Regarding the UINT16_MAX in those Kconfig, I think it is totally wrong. I approved that change though. I might submit a PR to restore it to `65535` or KB(64). This value has nothing to do with `UINT16_MAX` at all.

Comment (kartben) [reply boards/espressif/esp32c3_devkitc/Kconfig.defconfig:8]: @sylvioalves I agree it should be reverted -- and I should point out I also missed it during review :)

**Comment by kartben** [comment]
@raffarost just stumbled upon this but it also needs `esp32c3_usb.conf` / `esp32c3_usb.overlay` files to cover the boards that are using the USB variant of the SoC

The below fails otherwise, until `tests/drivers/adc/adc_api/socs/esp32c3_usb.conf` / `tests/drivers/adc/adc_api/socs/esp32c3_usb.overlay` are added.

```
west build -p -b esp32c3_luatos_core/esp32c3/usb tests/drivers/adc/adc_api -T drivers.adc
```

Could you please send a follow-up PR?

**Pull request merged**